wallet/src/actors/worker/methods.rs

```diff
@@ -135,6 +136,47 @@ impl Worker {
                 (id, default_account, Some(master_key))
             }
         };
+
+        let superblock_interval: i64 = 21;
+        let birth_date = match birth_date {
+            Some(epoch) => {
+                if is_old_epoch(
+                    epoch,
+                    self.params.epoch_constants.checkpoint_zero_timestamp,
+                    self.params.epoch_constants.checkpoints_period,
+                ) {
+                    return Err(Error::InvalidBirthDate(epoch));
+                }
+
+                let gen_fut = self.get_block_chain((epoch as i64) - superblock_interval, 2);
+                let gen_res: Vec<ChainEntry> = futures::executor::block_on(gen_fut)?;
+                let gen_entry = gen_res
+                    .get(0)
+                    .expect("It should always found a consolidated block for a validated epoch");
```

Comment:
```suggestion
                    .expect("It should always find a last consolidated block for a any epoch number");
```

(The block that you get may be the one for the provided epoch number, or the closest to that number that got consolidated, if there were rollbacks or no block candidates)